src/enqueue.cc

```diff
@@ -423,7 +423,11 @@ static ncclResult_t getAlgoInfo(struct ncclInfo* info, int collNetTypeSupport, i
     if (info->algorithm == NCCL_ALGO_COLLNET) nt += 3*WARP_SIZE;
   }
 #endif
-  info->nChannels = nc;
+  if (info->coll == ncclFuncAllToAllPivot) {
+    info->nChannels = comm->topo->pivotA2ANumChannels;
```

Comment:
We can only use up to established channel, thus need to take std::min from pivotA2ANumChannels and nc.

Reply:
Fixed.